core/modules/replica/Configuration.cc

```diff
@@ -39,79 +39,163 @@ namespace {
 
 LOG_LOGGER _log = LOG_GET("lsst.qserv.replica.Configuration");
 
+/**
+ * @param connectionUrl The connection URL.
+ * @param database The optional name of a database to replace the one defined in the url.
+ * @return The MySQL connection descriptor.
+ */
+database::mysql::ConnectionParams connectionParams(string const& connectionUrl, string const& database) {
+    database::mysql::ConnectionParams params = database::mysql::ConnectionParams::parse(connectionUrl);
+    if (!database.empty()) params.database = database;
+    return params;
+}
 } // namespace
 
 namespace lsst {
 namespace qserv {
 namespace replica {
 
+
 // These (static) data members are allowed to be changed, and they are set
 // globally for an application (process).
 bool         Configuration::_databaseAllowReconnect = true;
 unsigned int Configuration::_databaseConnectTimeoutSec = 3600;
 unsigned int Configuration::_databaseMaxReconnects = 1;
 unsigned int Configuration::_databaseTransactionTimeoutSec = 3600;
-string       Configuration::_qservMasterDatabasePassword = "";
-string       Configuration::_qservWorkerDatabasePassword = "";
+string       Configuration::_qservCzarDbUrl = "mysql://qsreplica@localhost:3306/qservMeta";
+string       Configuration::_qservWorkerDbUrl = "mysql://qsreplica@localhost:3306/qservw_worker";
```

Comment:
my experience is that port number does not force it to use TCP; IMO it would be better for my use case if the default was '127.0.0.1` (which does use TCP)

Reply:
The current implementation of the class `replica::DatabaseMySQL` forces TCP by:
```
    // Only allow TCP/IP, no UNIX sockets for now
    enum mysql_protocol_type const prot_type = MYSQL_PROTOCOL_TCP;
    mysql_options(_mysql, MYSQL_OPT_PROTOCOL, &prot_type);
```
Sockets are pretty much useless in Kubernetes due to various storage locality constraints.